modules/instrument_manager/php/instrument_manager.class.inc

```diff
@@ -112,62 +116,81 @@ class Instrument_Manager extends \NDB_Menu_Filter
      */
     protected function handlePOST(ServerRequestInterface $request): ResponseInterface
     {
-        if (!$this->canWriteFiles()) {
-            return (new \Zend\Diactoros\Response())
-                ->withStatus(403, 'Forbidden')
-                ->withBody(
-                    new \LORIS\Http\StringStream(
-                        json_encode(
-                            array('error' => CANNOT_WRITE_FILES)
-                        )
-                    )
-                );
-        }
         $uploaded_file = $request->getUploadedFiles()['install_file'];
         $filename      = $uploaded_file->getClientFilename();
         $instrument    = pathinfo($filename)['filename'];
+        // Do pre-processing before uploading files.
+        $err = '';
+        if (!$this->canWriteFiles()) {
+            $err    = self::CANNOT_WRITE_FILES;
+            $status = 503;
```

Comment:
This case shouldn't even come up really... I think I added this error handling a while ago but the module has changed a lot since then. The `projects/tables_sql/` and `projects/instruments` directories should always be writable by the Apache user. When https://github.com/aces/Loris/pull/4191 is merged it should take care of permissions checking and in that case this should really be a `500` error, or even a ConfigurationException.